MekHQ/src/mekhq/campaign/ai/utility/Memory.java

```diff
@@ -0,0 +1,67 @@
+package mekhq.campaign.ai.utility;
+
+import java.util.*;
+import java.util.function.Function;
+
+public class Memory {
```

Comment:
We definitely need a JavaDoc on this class as it's not immediately obvious what it does.